lib/kafka_producer.js

```diff
@@ -76,6 +97,20 @@ function KafkaProducer(options, callback) { // eslint-disable-line
             self.flushInterval = setInterval(flushCache, self.flushCycleSecs * 1000); // eslint-disable-line
         }
 
+        if (self.enableAudit) {
+            self.topicToMsgcntMap = {}; // map of topic name to current msg count
+
+            var produceAuditMsg = function doProduceAuditMsg() {
+                var auditMsg = self._generateAuditMsg();
+                if (auditMsg) {
+                    self.produce(self.auditTopicName, auditMsg, (Date.now() / 1000.0), callback);
```

Comment:
Remove the callback here or add your own. You shouldn't be using the user's callback for this.